### PR TITLE
Initial changes to allow user-functions to have <4 params, return <3 values

### PR DESCRIPTION
**Pull request by jlnav**
- [x] What's the simplest 1d_sampling test we could have now? :)
- [x] Documentation updates

Addresses #935

### CODE REVIEW

**Comment by jlnav** [comment]
```
@libE_fx
def sim_f(In):
    return np.linalg.norm(In)
```

I guess to try signifying that the user_f is a "simple" evaluator.

I try unpacking a matching "item" from calc_in, then packing the result(s) back into a new H after returning.

Somewhat more complicated solution than I envisioned since functions don't have access to their own specs in/out fields!

**Review by shuds13 — COMMENTED** [review]
Can we support dictionaries? Either different decorator name or overload.

**Comment by jlnav** [comment]
Based on discussion the further-simplification-decorator can come later. The ability to register functions that accept <=4 parameters is worth merging already,

File: libensemble/tests/regression_tests/test_1d_super_simple.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        "gen_f": gen_f,
+        "out": [("x", float, (1,))],
+        "user": {
+            "gen_batch_size": 500,
+            "lb": np.array([-3]),
+            "ub": np.array([3]),
+        },
+    }
+
+    persis_info = add_unique_random_streams({}, nworkers + 1, seed=1234)
```
Comment (shuds13) [inline libensemble/tests/regression_tests/test_1d_super_simple.py:51]: Why do we used fixed seed here?

File: libensemble/worker.py [hunk 2]
```diff
@@ -253,22 +253,27 @@ def _handle_calc(self, Work: dict, calc_in: npt.NDArray) -> (npt.NDArray, dict,
...
+                    return out
+                elif len(out) == 2:  # Out, persis_info OR Out, calc_status
+                    if isinstance(out[1], int) or isinstance(out[1], str):  # got Out, calc_status
+                        calc_status = out[1]
+                        return out[0], Work["persis_info"], calc_status
+                    return *out, calc_status  # got Out, persis_info
+                else:
+                    return out, Work["persis_info"], calc_status
+            else:
+                return np.zeros(1), Work["persis_info"], calc_status
```
Comment (shuds13) [inline libensemble/worker.py:275]: What happens with this zero? Should it be None?

File: libensemble/tests/regression_tests/test_1d_super_simple.py (hunk 1 above)
Comment (jmlarson1) [reply libensemble/tests/regression_tests/test_1d_super_simple.py:51]: From what I see, this is the only test where we fix a seed to `add_unique_random_streams`. So we are fixing the seed just to test that part of code.

File: libensemble/worker.py (hunk 2 above)
Comment (jlnav) [reply libensemble/worker.py:275]: Based on discussion, it probably _could_ be None, then the manager is responsible for slotting in the value

**Comment by jlnav** [comment]
- [ ] Check behavior of blank specs_out-fields and returning nothing from a function

**Pull request merged**